lib/awful/rules.lua

```diff
@@ -308,6 +309,32 @@ function rules.extra_properties.geometry(c, _, props)
     c:geometry(new_geo) --TODO use request::geometry
 end
 
+--- Create a new tag based on a rule.
+-- @tparam client c The client
+-- @tparam boolean|function|string value The value.
+-- @treturn tag The new tag
+function rules.high_priority_properties.new_tag(c, value)
+    local ty = type(value)
+    local t = nil
+
+    if ty == "boolean" then
+        -- Create a new tag named after the client class
+        t = a_tag.add(c.class or "N/A", {screen=c.screen})
```

Comment:
Is it a good idea to use `c.screen` here? A non-high-priority-rule could modify the screen later.